webserver/observe/metrics/src/main/java/io/helidon/webserver/observe/metrics/MetricsFeature4.java

```diff
@@ -0,0 +1,428 @@
+ * limitations under the License.
+ */
+package io.helidon.webserver.observe.metrics;
+
+//import java.lang.System.Logger.Level;
+//import java.util.Collections;
+//import java.util.List;
+//import java.util.Objects;
+//import java.util.Optional;
+//import java.util.ServiceLoader;
+//import java.util.Set;
+//import java.util.function.Supplier;
+//import java.util.stream.Stream;
+//
+//import io.helidon.common.HelidonServiceLoader;
+//import io.helidon.common.LazyValue;
+//import io.helidon.common.media.type.MediaType;
+//import io.helidon.common.media.type.MediaTypes;
+//import io.helidon.config.Config;
+//import io.helidon.config.metadata.ConfiguredOption;
+//import io.helidon.http.Http;
+//import io.helidon.metrics.api.KeyPerformanceIndicatorMetricsConfig;
+//import io.helidon.metrics.api.MeterRegistry;
+//import io.helidon.metrics.api.MeterRegistryFormatter;
+//import io.helidon.metrics.api.MetricsConfig;
+//import io.helidon.metrics.api.MetricsFactory;
+//import io.helidon.metrics.spi.MeterRegistryFormatterProvider;
+//import io.helidon.webserver.KeyPerformanceIndicatorSupport;
+//import io.helidon.webserver.http.Handler;
+//import io.helidon.webserver.http.HttpRouting;
+//import io.helidon.webserver.http.HttpRules;
+//import io.helidon.webserver.http.HttpService;
+//import io.helidon.webserver.http.ServerRequest;
+//import io.helidon.webserver.http.ServerResponse;
+//
+//import static io.helidon.http.Http.HeaderNames.ALLOW;
+//import static io.helidon.http.Http.Status.METHOD_NOT_ALLOWED_405;
+//import static io.helidon.http.Http.Status.NOT_ACCEPTABLE_406;
+//import static io.helidon.http.Http.Status.NOT_FOUND_404;
+//import static io.helidon.http.Http.Status.OK_200;
+
+/**
+ * Support for metrics for Helidon Web Server.
+ *
+ * <p>
+ * By defaults creates the /metrics endpoint.
+ * <p>
+ * To register with web server:
+ * <pre>{@code
+ * Routing.builder()
+ *        .register(MetricsSupport.create())
+ * }</pre>
+ * <p>
+ * This class supports finer grained configuration using Helidon Config:
+ * {link #create(io.helidon.config.Config)}. The following configuration parameters can be used:
+ * <table border="1">
+ * <caption>Configuration parameters</caption>
+ * <tr><th>key</th><th>default value</th><th>description</th></tr>
+ * <tr><td>helidon.metrics.context</td><td>/metrics</td><td>Context root under
+ * which the rest endpoints are available</td></tr>
+ * <tr><td>helidon.metrics.base.${metricName}.enabled</td><td>true</td><td>Can
+ * control which base metrics are exposed, set to false to disable a base
+ * metric</td></tr>
+ * </table>
+ * <p>
+ * The application metrics registry is then available as follows:
+ * <pre>{@code
+ *  req.context().get(MetricRegistry.class).ifPresent(reg -> reg.counter("myCounter").inc());
+ * }</pre>
+ */
+// TODO change back to extend HFS
+//public class MetricsFeature4 extends HelidonFeatureSupport {
+public class MetricsFeature4 /* extends MetricsFeature */ {
```

Comment:
Please remove this file

Reply:
Sigh. I was so careful to keep that out of my pushes for the longest time. Gone.